devicetypes/qubino/qubino-dimmer.src/qubino-dimmer.groovy

```diff
@@ -387,11 +387,21 @@ private dimmerEvents(physicalgraph.zwave.Command cmd, ep = null) {
 	def value = (cmdValue ? "on" : "off")
 	def result = [createEvent(name: "switch", value: value)]
 	if (cmdValue && cmdValue <= 100) {
+		cmdValue = adjustValueToRange(cmd.value)
 		result << createEvent(name: "level", value: cmdValue == 99 ? 100 : cmdValue)
 	}
 	return result
 }
 
+Integer adjustValueToRange(value){
+	if(value == 0){
+		return 0
+	}
+	def minDimmingLvlPref = settings.minimumDimmingValue ?: parameterMap.find({it.key == 'minimumDimmingValue'}).defaultValue
+	def adjustedValue = (((value - minDimmingLvlPref)/(100 - minDimmingLvlPref)) * 100) as Integer
+	return Math.max(adjustedValue, minDimmingLvlPref)
```

Comment:
I don't think you want to be doing this `math.max` anymore, right? If we're doing 50->55->50, presumably the user can still set the level to 1 even if the min dimming level is 10, right?

Reply:
I changed that and now user can set 1% level. I left Math.max with 1 instead of minDimmingLvlPref, because I wanted to be sure that after calculation we will not get any out of range values.